packages/discord.js/src/structures/User.js

```diff
@@ -151,6 +151,30 @@ class User extends Base {
     } else {
       this.avatarDecorationData = null;
     }
+
+    /**
+     * @typedef {Object} UserPrimaryGuild
+     * @property {?Snowflake} identityGuildId - The id of the user's primary guild
+     * @property {?boolean} identityEnabled - Whether the user is displaying the primary guild's server tag
+     * @property {?string} tag - The user's server tag. Limited to 4 characters
+     * @property {?string} badge - The server tag badge hash
+     */
+
+    if (data.primary_guild) {
+      /**
+       * The primary guild of the user
+       *
+       * @type {?UserPrimaryGuild}
+       */
+      this.primaryGuild = {
```

Comment:
Typically we'd also have some sort of getter for the `Guild` instance.
But since the name of this is `primaryGuild`... Not sure what to do about that

Reply:
Random thought: what if we make `PrimaryGuild` as a class which has these properties? Then we could include a getter to the guild.

Reply:
Would we then also duplicate the `guildTagBadgeURL` method there? It's currently directly on the `User` class